checker/check.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2018 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checker
+
+import (
+	authz "github.com/envoyproxy/data-plane-api/api/auth"
+
+	"github.com/projectcalico/app-policy/policystore"
+
+	"github.com/projectcalico/app-policy/proto"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/genproto/googleapis/rpc/code"
+	"google.golang.org/genproto/googleapis/rpc/status"
+	"strings"
+)
+
+var OK = code.Code_value["OK"]
+var PERMISSION_DENIED = code.Code_value["PERMISSION_DENIED"]
+
+type Action int
+
+const (
+	ALLOW Action = iota
+	DENY
+	LOG
+	PASS
+	NO_MATCH // Indicates policy did not match request. Cannot be assigned to rule.
+)
+
+// Check a list of policies and return OK if the check passes, or PERMISSION_DENIED if the check fails.
+// Note, if no policy matches, the default is PERMISSION_DENIED.
+func checkStore(store *policystore.PolicyStore, req *authz.CheckRequest) (s status.Status) {
+	s = status.Status{Code: PERMISSION_DENIED}
+	ep := store.Endpoint
+	if ep == nil {
+		log.Warning("CheckRequest before we synced Endpoint information.")
+		return
+	}
+	if len(ep.Tiers) > 0 {
+		log.Debug("Checking policy tier 1.") // We only support a single tier.
+
+		tier := ep.Tiers[0]
+		policies := tier.IngressPolicies
+	Policy:
+		for i, name := range policies {
+			pID := proto.PolicyID{Tier: tier.GetName(), Name: name}
+			policy := store.PolicyByID[pID]
+			action := checkPolicy(policy, req)
+			log.Debugf("Policy %d %v returned action %v", i, pID, action)
+			switch action {
+			case NO_MATCH:
+				continue
+			// If the Policy matches, end evaluation (skipping profiles, if any)
+			case ALLOW:
+				s.Code = OK
+				return
+			case DENY:
+				s.Code = PERMISSION_DENIED
+				return
+			case PASS:
+				// Pass means end evaluation of policies and proceed to profiles, if any.
+				break Policy
+			case LOG:
+				panic("policy should never return LOG action")
+			}
+		}
+	}
+	// If we reach here, there were either no policies, or none that matched.
+	if len(ep.ProfileIds) > 0 {
+		for i, name := range ep.ProfileIds {
```

Comment:
if you don't need the index in the loop, then I suggest you `_` it like `for _, name := range ep.ProfileIDs` (seems like it's only used for a debug message, which might not be needed if you're already logging the name)

Reply:
I do want to log it because policies and profiles are not explicitly ordered when the user configures them, so it will be helpful to explicitly see the ordering in the logs.